ndmg/scripts/ndmg_dwi_pipeline.py

```diff
@@ -183,107 +184,102 @@ def ndmg_dwi_worker(
-            os.mkdir(namer.dirs["tmp"]["reg_m"])
-        except:
-            pass
+
+    # define registration directory locations
+    # TODO: possibly just pull these from a container generated by `gen_utils.make_initial_directories`
+    reg_dirs = ["anat/preproc", "anat/registered", "tmp/reg_a", "tmp/reg_m"]
+    reg_dirs = [outdir / loc for loc in reg_dirs]
+    prep_anat, reg_anat, tmp_rega, tmp_regm = reg_dirs
+
+    if not skipreg:
+        for dir_ in [prep_anat, reg_anat]:
+            if gen_utils.has_files(dir_):
+                gen_utils.as_directory(dir_, remove=True)
+        if gen_utils.has_files(tmp_rega) or gen_utils.has_files(tmp_regm):
+            for tmp in [tmp_regm, tmp_rega]:
+                gen_utils.as_directory(tmp, remove=True)
 
     # Check orientation (t1w)
     start_time = time.time()
-    t1w = gen_utils.reorient_img(t1w, namer)
-    t1w = gen_utils.match_target_vox_res(t1w, vox_size, namer, sens="t1w")
+    t1w = gen_utils.reorient_t1w(t1w, prep_anat)
+    t1w = gen_utils.match_target_vox_res(t1w, vox_size, outdir, sens="anat")
 
     print("Running registration in native space...")
 
     # Instantiate registration
     reg = register.DmriReg(
-        namer, nodif_B0, nodif_B0_mask, t1w, vox_size, skull, simple=False
+        outdir, nodif_B0, nodif_B0_mask, t1w, vox_size, skull, simple=False
     )
 
     # Perform anatomical segmentation
-    if (skipreg is True) and os.path.isfile(reg.wm_edge):
+    if skipreg and os.path.isfile(reg.wm_edge):
         print("Found existing gentissue run!")
     else:
         reg.gen_tissue()
 
     # Align t1w to dwi
-    if (
-        (skipreg is True)
-        and os.path.isfile(reg.t1w2dwi)
-        and os.path.isfile(reg.mni2t1w_warp)
-        and os.path.isfile(reg.t1_aligned_mni)
-    ):
+    t1w2dwi_align_files = [reg.t1w2dwi, reg.mni2t1w_warp, reg.t1_aligned_mni]
+    existing_files = all(map(os.path.isfile, t1w2dwi_align_files))
+    if skipreg and existing_files:
         print("Found existing t1w2dwi run!")
     else:
         reg.t1w2dwi_align()
 
     # Align tissue classifiers
-    if (
-        (skipreg is True)
-        and os.path.isfile(reg.wm_gm_int_in_dwi)
-        and os.path.isfile(reg.vent_csf_in_dwi)
-        and os.path.isfile(reg.corpuscallosum_dwi)
-    ):
+    tissue_align_files = [
+        reg.wm_gm_int_in_dwi,
+        reg.vent_csf_in_dwi,
+        reg.corpuscallosum_dwi,
+    ]
+    existing_files = all(map(os.path.isfile, tissue_align_files))
+    if skipreg and existing_files:
         print("Found existing tissue2dwi run!")
     else:
         reg.tissue2dwi_align()
 
     # Align atlas to dwi-space and check that the atlas hasn't lost any of the rois
-    labels_im_file_list = reg_utils.skullstrip_check(
-        reg, labels, namer, vox_size, reg_style
-    )
+    skullstrip_files = [reg, parcellations, outdir, prep_anat, vox_size, reg_style]
+    labels_im_file_list = reg_utils.skullstrip_check(*skullstrip_files)
```

Comment:
was a bit too long as just one line and black uglified it, but also maybe overkill to define a whole variable for just this purpose

maybe in the future use `_` to signify a one-time-use temp variable in situations like this

Reply:
88d589d